source/frontend/src/components/propertySelector/search/PropertySelectorSearchContainer.tsx

```diff
@@ -174,9 +201,22 @@ export const PropertySelectorSearchContainer: React.FC<IPropertySelectorSearchCo
       });
 
       // match the region and district for all found properties
-      await Promise.all(
-        propertyResults.map(p => matchRegionAndDistrict(p, findRegion, findDistrict)),
-      );
+      const regionDistrictTasks = getRegionAndDisctricts(propertyResults, findRegion, findDistrict);
+
+      const regionDistrictResults: Dictionary<RegionDistrictResult> = {};
+      for (const latLngKey in regionDistrictTasks) {
```

Comment:
This syntax is easier to read/more concise in my OP: 
dictionary.forEach((value, key)

Reply:
This may be premature optimization, but this implementation seems to loop through the list of properties 4 times do to this processing - how does this perform on large lists of properties?